doc/releases/changelog-dev.md

```diff
@@ -170,6 +173,10 @@
 
 <h3>Documentation 📝</h3>
 
+* The Catalyst Runtime Device Interface page has been updated, pointing to the
+  `QuantumDevice` struct documentation. Also the QIR C-API page has been removed.
```

Comment:
Some guidance on changelog entries: They are as much about the "why" as about the "what", and they should also answer the question "so what" (i.e. what is the impact on the user?) unless that's obvious (less relevant here).

In this case, the entry is missing information on why we removed it. For instance, did we just streamline the docs, or was it removed because the relevant code had been deprecated?
```suggestion
* The API section in the documentation has been simplified. The Catalyst 'Runtime Device Interface'
  page has been updated to point directly to the documented `QuantumDevice` struct, and the 'QIR
  C-API' page has been removed due to limited utility.
```